powerlevel9k.zsh-theme

```diff
@@ -1004,10 +1004,30 @@ prompt_ssh() {
 # Status: return code if verbose, otherwise just an icon if an error occurred
 set_default POWERLEVEL9K_STATUS_VERBOSE true
 set_default POWERLEVEL9K_STATUS_OK_IN_NON_VERBOSE false
+set_default POWERLEVEL9K_STATUS_SHOW_PIPESTATUS false
 prompt_status() {
-  if [[ "$RETVAL" -ne 0 ]]; then
+  local ec_text
+  local ec_sum
+  local ec
+
+  if [[ $POWERLEVEL9K_STATUS_SHOW_PIPESTATUS == true ]]; then
+    ec_text=${RETVALS[1]}
+    ec_sum=${RETVALS[1]}
+
+    for ec in "${(@)RETVALS[2,-1]}"; do
+      ec_text="${ec_text}|${ec}"
+      ec_sum=$(( $ec_sum + $ec ))
+    done
+  else
+    # We use RETVAL instead of the right-most RETVALS item because
+    # PIPE_FAIL may be set.
+    ec_text=${RETVAL}
+    ec_sum=${RETVAL}
+  fi
+
+  if (( ec_sum > 0 )); then
```

Comment:
The exit codes should _never_ be negative.  It is by definition an unsigned 8-bit int. The first 7 bits are for "normal" exit codes (0–127).  If the 8th bit is set then the process was killed by an exit code  For example:

``` sh
(sleep 10 ; a=$? ; echo "\$?: $a   8th bit: $(( a >> 7 ))   low bits $(( a & 127 ))" ) & sleep 1 ; kill -9 $(pidof sleep)
[1] 30696
$?: 137   8th bit: 1   low bits 9
[1]  + 30696 done       ( sleep 10; a=$? ; echo ; )
```

In the example:

- `$?` was 137
- The 8th bit was set.
- The low bits were `9`
- If we used a signed 8-bit int, this would be `-9` but shells don't work that way.

I couldn't find the `zsh` references for `$?` but I'm pretty sure it follow the posix standard.

I would be interested in finding out I'm wrong:  Do you have a command that returns a less-than-zero exitcode?